src/IBM.Watson.Assistant.v1/AssistantService.cs

```diff
@@ -96,17 +97,16 @@ public AssistantService(IClient httpClient) : base(SERVICE_NAME, URL)
         /// entities, and context from the last response. (optional)</param>
         /// <param name="nodesVisitedDetails">Whether to include additional diagnostic information about the dialog
         /// nodes that were visited during processing of the message. (optional, default to false)</param>
-        /// <param name="customData">Custom data object to pass data including custom request headers.</param>
         /// <returns><see cref="MessageResponse" />MessageResponse</returns>
-        public MessageResponse Message(string workspaceId, MessageRequest request = null, bool? nodesVisitedDetails = null, Dictionary<string, object> customData = null)
+        public DetailedResponse<MessageResponse> Message(string workspaceId, JObject input = null, List<JObject> intents = null, List<JObject> entities = null, bool? alternateIntents = null, JObject context = null, JObject output = null, bool? nodesVisitedDetails = null)
```

Comment:
This line prompted a question that I probably should've had in the past since it used to be similar, but do we need to be generating models like `MessageInput`? As far as I can tell that's the type that `input` corresponds to, but you're accepting a `JObject`. I also saw before that it was `dynamic`.

Reply:
well we still can use the `MessageInput` class and convert it into a JObject if they want to use the explicitly set properties. 

```
MessageInput input = new MessageInput()
{
    Text = inputString
};

var results = service.Message(
    workspaceId: workspaceId,
    input: JObject.FromObject(input)
    );
```

They can also set arbitrary fields
```
MessageInput input = new MessageInput()
{
    Text = inputString
};

JObject inputJobj = JObject.FromObject(input);
inputJobj["foo"] = "bar";

var results0 = service.Message(
    workspaceId: workspaceId,
    input: inputJobj 
    );
```

The `dynamic` issue is a bit different. In generating the Unity SDK I found that .NET could only have one `dynamic` parameter per method. This was never an issue since Unity and .NET were never exploding body params and never needed to have more than one `dynamic` param in the method signature. I found a different way that works (JObject) in the Json library we are using.

@padamstx is looking at reworking this in the Java SDK and the fix might be applicable here. That would come in a separate PR.

Reply:
Sounds good 👍  I figured you had a good reason but wanted to confirm.